### ISSUE TITLE
Source File: improve error messages & documentation

### ISSUE THREAD

**Issue by clnoll** [airbytehq/airbyte#33061]
A customer had two issues using the File connector that caused them to contact Support, that could have been avoided if we'd had better error messaging.

In the first case, the connection check succeeded, but the sync resulted in no records being sent. On closer inspection, the file was being blocked by a warning stating that the file is too big to scan for viruses; this warning message was actually turned into the catalog's schema. Since no records matched it, nothing was emitted from the sync. 

Once they broke the file into smaller parts, they got configuration errors preventing them from setting up the connector. However, the error message didn't give them any hint as to the problem. It turned out that the file was encoded with `utf-16` (we assume `utf-8`) all they needed to do was add an `{"encoding": "utf-16"}` to the Reader Options section of the connector during setup.  Internally, our error message showed that the problem was with the encoding, but the user had no way of knowing that this was the problem. 

To address these we should:
1. Update documentation for source-file describing the warning message and how to determine whether it will block your file from being downloaded. We should note that there is a setting that users may set that will remove the warning for them, making it look like the problem is fixed, but that doesn't guarantee that the warning won't block `source-file` from being able to download it.
2. Look for the warning and raise an exception if it's hit, instead of making it part of the schema.
3. Surface more information on errors during the connectivity check. For encoding-related errors, suggest encodings that they can try and how to put them into the Reader Options.
**Not in scope**: Use a library for determining the encoding (e.g. [`chardet`](https://github.com/chardet/chardet)). A separate ticket will be created to consider integrating this with the file-based CDK. Since `source-file` will ultimately be migrated to use the file-based CDK, it's probably not worthwhile to update it to detect encoding too.

Closes https://github.com/airbytehq/oncall/issues/3513.
Slack discussion [here](https://airbytehq-team.slack.com/archives/C02U9R3AF37/p1699378156800829).
